osquery/tables/system/linux/md_tables.cpp

```diff
@@ -0,0 +1,735 @@
+  size_t n = 0;
+
+  if (lines.size() < 1) {
+    return;
+  }
+
+  // This should always evaluate to true, but just in case we check.
+  if (lines[0].find("Personalities :") != std::string::npos) {
+    std::string pline(lines[0].substr(sizeof("Personalities :") - 1));
+    parseMDPersonalities(pline, result.personalities);
+
+    n = 1;
+
+  } else {
+    LOG(WARNING) << "mdstat Personalites not found at line 0: " << lines[0];
+  }
+
+  while (n < lines.size()) {
+    if (lines[n].find_first_not_of("\t\r\v ") == std::string::npos) {
+      n += 1;
+      continue;
+    }
+    // Work off of first 2 character instead of just the first to be safe.
+    std::string firstTwo = lines[n].substr(0, 2);
+    if (firstTwo == "md") {
+      auto mdline(split(lines[n], ":", 1));
+      if (mdline.size() < 2) {
+        LOG(WARNING) << "Unexpected md device line structure: " << lines[n];
+        continue;
+      }
+
+      MDDevice mdd;
+      mdd.name = mdline[0];
+      boost::algorithm::trim(mdd.name);
+
+      auto settings(split(mdline[1], " "));
+      trimStrs(settings);
+      // First 2 of settings are always status and RAID level
+      if (settings.size() >= 2) {
+        mdd.status = settings[0];
+        mdd.raidLevel = settings[1];
+
+        for (size_t i = 2; i < settings.size(); i++) {
+          mdd.drives.push_back(parseMDDrive(settings[i]));
+        }
+      }
+
+      /* Next line is device config and settings.  We handle here instead of
+       * later b/c pieces are need for both md_drives and md_devices table */
+      auto configline(split(lines[n + 1]));
+      if (configline.size() < 4) {
+        LOG(WARNING) << "Unexpected md device config: " << lines[n + 1];
+
+      } else {
+        trimStrs(configline);
+        // mdd.usableSize = configline[0] + " " + configline[1];
+        if (configline[1] == "blocks") {
+          mdd.usableSize = std::stoll(configline[0]);
+        } else {
+          LOG(WARNING) << "Did not find size in mdstat for " << mdd.name;
+        }
+
+        mdd.healthyDrives = configline[configline.size() - 2];
+        mdd.driveStatuses = configline[configline.size() - 1];
+
+        if (configline.size() > 4) {
+          for (size_t i = 2; i < configline.size() - 2; i++) {
+            mdd.other += (" " + configline[i]);
+            boost::algorithm::trim(mdd.other);
+          }
+        }
+      }
+      // Skip config line for next iteration
+      n += 1;
+
+      // Handle potential bitmap, recovery, and resync lines
+      std::size_t pos;
+      while (true) {
+        if ((pos = lines[n + 1].find("recovery =")) != std::string::npos) {
+          std::string recovery(
+              lines[n + 1].substr(pos + sizeof("recovery =") - 1));
+          boost::algorithm::trim(recovery);
+          parseMDAction(recovery, mdd.recovery);
+          // Add an extra line for next iteration if so..
+          n += 1;
+
+        } else if ((pos = lines[n + 1].find("resync =")) != std::string::npos) {
```

Comment:
While the assignment-in-expression is done correctly here, :) it still scares me. Is there a way to refactor this block, maybe into another method, where the assignments can live outside of the conditional checking?